wurst/event/DamageModify.wurst

```diff
@@ -0,0 +1,462 @@
+		state = DamageState.FIRING
+		let temp = dealingInstance
+		dealingInstance = this
+		Log.trace("deal d" + I2S(this castTo int))
+		UnitDamageTarget(source, target, amount, attack, ranged, attackType, damageType, weaponType)
+		Log.trace("completed d" + I2S(this castTo int))
+		dealingInstance = temp
+		if state != DamageState.FIRED
+			Log.error("Damage.deal() returned not fully processed damage! (state: " + state.toString() + ")")
+			if not isReturned
+				destroy this
+			
+	// Event modification
+
+	/** Make actual amount leave target at at least 0.406 hp left */
+	function preventDeath()
+		doPreventDeath = true
+
+	/** Nullifies actual amount no matter what. */
+	function deflect()
+		doDeflect = true
+
+	/** Increases `amount` of the damage by `initialAmount * magnitude` */
+	function addMultiplier(real magnitude)
+		if state != DamageState.FIRED and state != DamageState.LISTENING
+			amount += initialAmount * (magnitude - 1)
+		else
+			Log.debug("Damage.addMultiplier() is used in the wrong damage state:" + state.toString())			
+
+	/** Increases `amount` of the damage with given amount */
+	function addAmount(real amount)
+		if state != DamageState.FIRED and state != DamageState.LISTENING
+			this.amount += amount
+		else
+			Log.debug("Damage.addAmount() is used in the wrong damage state:" + state.toString())			
+
+	/** Sets `amount` of the damage to given amount */
+	function setAmount(real amount)
+		if state != DamageState.FIRED and state != DamageState.LISTENING
+			this.amount = amount
+		else
+			Log.debug("Damage.setAmount() is used in the wrong damage state:" + state.toString())
+
+	/** Sets the amount of times that each DamageModifier is applied to this damage.
+		Should be in range [0, 32]. */
+	function setModifyCount(int val)
+		if state != DamageState.FIRED and state != DamageState.LISTENING
+			if val >= 0 and val <= 32
+				modifyCount = val
+			else
+				Log.debug("Damage.setModifyCount() is used with wrong value: " + val.toString())
+		else
+			Log.debug("Damage.setModifyCount() is used in the wrong damage state:" + state.toString())
+				
+
+// Handler of the event
+
+let queue = new LinkedList<Damage>
+var recursionAmount = 0
+var needTimer = false
+var dequeueRunning = false
+timer nullTimer = null
+
+function dequeue()
+	if dequeueRunning
+		return
+	if nullTimer != null
+		nullTimer.release()
+		nullTimer = null
+	dequeueRunning = true
+	while true
+		let e = queue.dequeue()
+		if e == null
+			break
+		e.invoke()
+	dequeueRunning = false
+	recursionAmount = 0
+	needTimer = false
+
+function retrieve() returns Damage
+	var d = dealingInstance
+	if d == null or d.getSource() != GetEventDamageSource() or d.getTarget() != GetTriggerUnit()
+		if d != null
+			Log.debug("Damage.handler() was ignored a damage due to incorrect target/source!")
+	else
+		dealingInstance = null // let no one else retrieve the same instance
+	return d
+
+function handler() returns bool
+	let damage = GetEventDamage()
+	if damage == 0
```

Comment:
Is there a reason your explicitly ignore 0 damage events? Could you make this a configurable bool `IgnoreZeroDamageEvents`, rather than make it hardcoded functionality?

Reply:
Removed hardcoded functionality, inserted the check into `getDamage_(From)_Event` and also encapsulated Damage data check inside retrieve into a separate configurable function.